### PR TITLE
frontend testing framework

### PR DESCRIPTION
**Pull request by tristanhcole**
* nuking old test setup
* updating package.json to correctly setup Jest and Enzyme
* adding frontend tests runner to `config.yml`

### CODE REVIEW

**Comment by tristanhcole** [comment]
[Frontend Web testing](https://trello.com/c/mlBIryaj/161-frontend-web-testing)

File: .circleci/config.yml [hunk 1]
```diff
@@ -141,6 +142,28 @@ jobs:
...
+          keys:
+            # when lock file changes, use increasingly general patterns to restore cache
+            - node-v1-{{ .Branch }}-{{ checksum "~/project/app/package-lock.json" }}
+            - node-v1-{{ .Branch }}-
+            - node-v1-
+      - run:
+          name: Run Frontend Tests
+          command: |
+            cd app
+            npm run test
```
Comment (tristanhcole) [inline .circleci/config.yml]: Split out front end tests from back end so CI runs faster
https://circleci.com/docs/2.0/parallelism-faster-jobs/

Comment (EDsCODE) [reply .circleci/config.yml]: parallelizing looks good..also i think the status check for "ci/circleci: test" needs to be removed

Comment (tristanhcole) [reply .circleci/config.yml]: good point - will leave active until this PR is ready to merge in - so that the other branches keep working!

Comment (EDsCODE) [reply .circleci/config.yml]: sounds good! will give it a full look this weekend

File: app/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// jest.config.js
+module.exports = {
+  verbose: true,
+  roots: [
+    "./client"
+  ],
+  transform: {
+    "^.+\\.[t|j]sx?$": "babel-jest"
```
Comment (EDsCODE) [inline app/jest.config.js]: we should use "ts-jest" here to transform tsx elements. we might need more coverage on the regex for ts/js elements too but either way ts-jest will allow us to test on tsx files

we can either do it here or i can add it in after the fact when i start writing tests in the next PR!

File: app/package.json [hunk 3]
```diff
@@ -62,20 +52,19 @@
...
-    "enzyme-adapter-react-16": "^1.1.1",
-    "enzyme-to-json": "^3.3.4",
+    "enzyme": "^3.10.0",
+    "enzyme-adapter-react-16": "^1.15.1",
+    "enzyme-to-json": "^3.4.3",
     "husky": "^4.2.3",
-    "identity-obj-proxy": "^3.0.0",
-    "jest": "^23.4.1",
-    "jest-cli": "^23.4.1",
+    "jest": "^24.0.0",
```
Comment (EDsCODE) [inline app/package.json]: could update these versions to latest/later versions unless there's a specific dependency intended

File: app/jest.config.js (hunk 2 above)
Comment (tristanhcole) [reply app/jest.config.js]: good point - updated https://kulshekhar.github.io/ts-jest/user/config/#jest-config-with-helper

> `ts-jest/presets/js-with-babel` | TypeScript files will be handled by ts-jest, and JavaScript files will be handled by babel-jest.

File: app/package.json (hunk 3 above)
Comment (tristanhcole) [reply app/package.json]: updated all jest related packages to latest

**Pull request merged**